lambda/sigterm.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
nit: would it make sense to move to a sub-package `signals`?

Reply:
This would require making the enableSIGTERM helper public, which would make it easy to mistakenly use the extension registration APIs before the first invoke, which I believe results in API errors.